object_store_factory/src/lib.rs

```diff
@@ -126,14 +126,20 @@ pub async fn build_object_store_from_opts(
             aws::add_amazon_s3_specific_options(url, &mut s3_options).await;
             aws::add_amazon_s3_environment_variables(&mut s3_options);
 
-            let (store, _) = parse_url_opts(url, s3_options)?;
+            let (mut store, _) = parse_url_opts(url, s3_options)?;
+            if !url.path().is_empty() {
+                store = Box::new(PrefixStore::new(store, url.path()));
+            }
```

Comment:
We should be able to just use the path returned by the object_store crate here and below.
```suggestion
            let (mut store, path) = parse_url_opts(url, s3_options)?;
            if !path.is_empty() {
                store = Box::new(PrefixStore::new(store, path));
            }
```

Reply:
That path is of type `object_store::path::Path` which doesn't have an `is_empty` method. I would stick to url.path() as it's what we use further up in L115